### DISCUSSION TITLE
time dependent neumann boundary condition

### DISCUSSION THREAD

**Discussion by Abhishekc95** [lululxvi/deepxde#1793] [General]

**Comment by Abhishekc95** [comment]
@praksharma If you have a moment, I would appreciate your help

**Comment by praksharma** [comment]
I think your implementation is okay. It just that the baseline PINN really struggles with wave equation. You can look into papers like learning rate annealing PINN, or neural tangent kernel PINN. there are more works, such as time windows scheduling which is done in NVIDIA modulus to specifically solve such problems.

**Reply by Abhishekc95** [reply]
what about the case for inverse problem for wave equation. does baseline pinn work for that? or should i consider your recommendations for that as well?

**Comment by praksharma** [comment]
read this paper. yes baseline PINN works with wave equation, but needs a lot of tinkering due to the imbalanced loss terms.
https://arxiv.org/abs/2006.11894

**Reply by Abhishekc95** [reply]
@praksharma i need some more help. im trying to solve inverse wave equation. I have simulated data from fem, where i have displacement data at 4 time steps. how  can input this to solve for constant c in the wave equation?. i looked at other inverse problems in issues, i found people using func( some exact solution).

**Reply by praksharma** [reply]
You can set your sparse data as BCs, although they are not BC but NNs are optimisers and they don't care about the physics.
Use PointSetBC.
https://github.com/lululxvi/deepxde/blob/master/deepxde/icbc/boundary_conditions.py

**Reply by Abhishekc95** [reply]
just a follow up question to this. what should input in anchors, like i mentioned in previous reply i have u at t1,t2. The inverse examples have only 1 time shot. im using pointsetBC for observations.
